<?xml version="1.0" ?>
<launch>

        <arg name="qcID"/>

	<arg name="rotortype" value="UAV_Octo" />
        <arg name="landHeight" value="-0.32" />
        <arg name="initialMass" value="0.9" />
        <arg name="takeOffHeight" value="-1.00" />
        <!-- setting the vicon topic for the state estimation -->
        <arg name="vicon_topic" default="/TeleKyb/Vicon/$(arg rotortype)_$(arg qcID)/$(arg rotortype)_$(arg qcID)" />


        <node name="TeleKybCore_$(arg qcID)" pkg="telekyb_core" type="telekyb_core_node">
                <param name="TeleKybCore/tRobotID" value="$(arg qcID)" />
                <param name="StateEstimatorController/tPluginLookupName" value="state_estimators_plugin::ViconStateEstimator" />
                <param name="TrajectoryController/tPluginLookupName" value="trajectory_trackers_plugin::StandardTrajectoryTracker" />
                <param name="StandardTrajectoryTracker/tCommandsTopic" value="/TeleKyb/$(arg qcID)/Commands" />

                <param name="StateEstimatorController/tTransformParentID" value="/world_link" />
                <param name="StateEstimatorController/tTransformChildFrame" value="/base_link_robot_$(arg qcID)" />
                <param name="StateEstimatorController/tPublishRosTransform" value="true" />
                
		<param name="ViconStateEstimator/tViconSeTopicName" value="$(arg vicon_topic)" />

                <param name="StandardMassEstim/tInitialMass" value="$(arg initialMass)"/>

                <param name="PositionControl/tDerGainVelMode" value="5" /> <!-- value of 5 seems quite nice together with the values below -->
                <param name="PositionControl/tPropGain" value="2.25" /><!-- default: 9 -->
                <param name="PositionControl/tDerivGain" value="3.75" /><!-- default: 7.5 -->
                <param name="PositionControl/tIntegGain" value="0" /> <!-- default: 4.5 -->

                <param name="PositionControl/tMaxRollSin" value="0.087155743" /><!-- sin(5.0*M_PI/180.0) -->
                <param name="PositionControl/tMaxPitchSin" value="0.087155743" /><!-- sin(5.0*M_PI/180.0) -->

                <param name="TrajectoryProcessorController/tTrajectoryModules" value='[]' />
                <param name="StandardMassEstim/tInitialMass" value="$(arg initialMass)" />
                <param name="tCenterValueX" value="512" />
                <param name="tCenterValueY" value="512" />
                <param name="tTakeOffDestination" value="[0.0, 0.0, $(arg takeOffHeight)]" />
                <param name="tLandDestinationHeight" value="$(arg landHeight)" />

                <param name="tObsPointsTopicName" value="/TeleKyb/Obstacle_$(arg qcID)/ObstaclePoints"/>

                <!--param name="tJoystickUseDeadManSwitch" value="false" / -->
                <!--param name="tJoystickUsePositionMode" value="false" / -->
        </node>

</launch>
